src/mame/video/plygonet.cpp

```diff
@@ -18,67 +18,44 @@
 
 TILE_GET_INFO_MEMBER(polygonet_state::ttl_get_tile_info)
 {
-	int attr, code;
-
-	code = m_ttl_vram[tile_index]&0xfff;
-
-	attr = m_ttl_vram[tile_index]>>12;  /* palette in all 4 bits? */
+	uint16_t *ttl_vram = (uint16_t *)&m_ttl_vram[0];
+	int code = ttl_vram[tile_index] & 0xfff;
+	int attr = ttl_vram[tile_index] >> 12;  /* palette in all 4 bits? */
```

Comment:
Because you’re doing a `u32` to `u16` reinterpret cast here, you need to consider host vs guest endianness (yes, this is a pre-existing bug).  Change the code here to:
```
	uint16_t *ttl_vram = (uint16_t *)&m_ttl_vram[0];
	int code = ttl_vram[BYTE_XOR_BE(tile_index)] & 0xfff;
	int attr = ttl_vram[BYTE_XOR_BE(tile_index)] >> 12;  /* palette in all 4 bits? */
```

Then get rid of the `scan_rows` and `scan_cols` mapper functions, and use the standard `TILEMAP_SCAN_ROWS` and `TILEMAP_SCAN_COLS` mapper functions.